include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,372 @@
+    }
+
+    //! @brief Resizes the container to contain count elements.
+    //!
+    //! If the current size is greater than `n`, the container is reduced to its first `n` elements.
+    //! If the current size is less than `n`, additional elements are appended. Those elements are
+    //! either default-inserted or copied if `value` is provided.
+    //! @param n      new size of the container
+    //! @param value  the value to initialize the new elements with
+    void resize(std::size_t n, value_type value = {})
+    {
+      auto sz = size();
+      if(n <= capacity())
+      {
+        storage.size_ = n;
+      }
+      else
+      {
+        // Grow twice per resize
+        soa_vector that( storage_t(storage.get_allocator(),n,2*n) );
+        eve::algo::copy(*this, that.begin_aligned());
+        this->operator=(std::move(that));
+      }
+
+      // If needed, fill with new data
+      if(n > sz)
+        eve::algo::fill(algo::as_range(begin() + sz, begin()+n), value );
+    }
+
+    //! @brief Exchanges the contents of the container with those of `other`.
+    //! Does not invoke any move, copy, or swap operations on individual elements.
+    //! All iterators and references remain valid. The past-the-end iterator is invalidated.
+    void swap(soa_vector& other) noexcept
+    {
+      storage.swap(other.storage);
+    }
+
+    //! @brief Swaps the contents of `lhs` and `rhs` by calling `lhs.swap(rhs)`.
+    friend EVE_FORCEINLINE void swap(soa_vector &lhs, soa_vector &rhs) noexcept
+    {
+      lhs.swap(rhs);
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    public:
+    //==============================================================================================
+    //! @name Element access
+    //! @{
+    //==============================================================================================
+    //! Returns a zipped aligned pointer to the beginning
+    EVE_FORCEINLINE auto data_aligned()         { return storage.data_aligned(); }
+
+    //! Returns a zipped aligned pointer to the beginning
+    EVE_FORCEINLINE auto data_aligned()  const  { return storage.data_aligned(); }
+
+    //! Returns a zipped pointer to the beginning
+    EVE_FORCEINLINE auto data()       { return  storage.data(); }
+
+    //! Returns a constant zipped pointer to the beginning
+    EVE_FORCEINLINE auto data() const { return  storage.data(); }
+
+    //! @brief Returns the value of the `i`th element of the container
+    //! @param i Index of the value to retrieve
+    EVE_FORCEINLINE auto get(std::size_t i) const { return eve::read(begin() + i); }
+
+    //! @brief Modify the value of the `i`th element of the container
+    //! @param i Index of the value to write
+    //! @param v Value to write
+    EVE_FORCEINLINE void set(std::size_t i, value_type const& v)
```

Comment:
By value? Ah w/e. Probably use size_type typedef though.

Reply:
If the value_type is a really large struct (like I dunno 5-6 members) it'll be better to pass by reference me think